### ISSUE TITLE
Universal morphisms for (co)limits in FinSet

### ISSUE THREAD

**Issue by epatters** [AlgebraicJulia/Catlab.jl#238]
We should finish up limits and colimits in FinSet by implementing the universal property, i.e., adding functions to construct the universal morphisms. For (co)limits of specific shape, we should use the standard names, such as `pair` and `copair`, and we should also support a generic function for arbitrary (co)limits, say `factorize`. See `src/theories/Limits.jl` for how this might look.

Following #235, the limits can just have standard type `Limit`, but for the colimits it will probably be helpful to retain the union find as an extra field:

```julia
struct FinSetIntColimit{Diagram,Cocone<:Multispan{FinSet{Int}}} <: AbstractColimit{FinSet{Int},Diagram}
  diagram::Diagram
  cocone::Cocone
  sets::IntDisjointSets
end
```

**Comment by jpfairbanks** [comment]
Just got bitten by this in implementing LinRel. Definite need!

**Comment by epatters** [comment]
Resolved by #269.
